engine/action_transaction.go

```diff
@@ -360,7 +361,11 @@ func (c *Client) HandleTxCallback(ctx context.Context, callbackResp *chainmodels
 	}
 
 	tx.BlockHash = callbackResp.BlockHash
-	tx.BlockHeight = uint64(callbackResp.BlockHeight)
+	if callbackResp.BlockHeight < 0 {
+		return fmt.Errorf("invalid block height: %d", callbackResp.BlockHeight)
```

Comment:
In `engine` package, please use `spverrors.Wrapf / Newf` or return defined SPVError